src/controllers/water-level-changes.js

```diff
@@ -0,0 +1,132 @@
+const QUERY_BASE = `${URL}?$query=`
+
+/**
+ * Water Level Changes (North Saskatchewan River at Edmonton)
+ */
+module.exports = async function(req, res) {
+  console.log('Water Level Changes Controller')
+  let storedData = await req.cache.getLatest(KEY)
+  let storedUpdated
+  if (storedData) storedUpdated = storedData.last_updated
+
+  let getUpdatedQuery = `${QUERY_BASE}
+    SELECT date_and_time, water_level_m
+    WHERE station_number = '05DF001'
+    ORDER BY date_and_time DESC
+    LIMIT 1`
+  let updatedData
+  let recentUpdated
+  let recentWaterLevel
+  try {
+    let rawJsonUpdated = await request(getUpdatedQuery)
+    updatedData = JSON.parse(rawJsonUpdated)
+    recentUpdated = updatedData[0].date_and_time
+    recentWaterLevel = updatedData[0].water_level_m
+  } catch (e) {
+    e.code = 500
+    throw e
+  }
+
+  let responseValues
+  if (storedUpdated && new Date(storedUpdated) >= new Date(recentUpdated)) {
+    console.log('No new updates')
+    responseValues = await req.cache.getAll(KEY, req.body['limit'])
+  } else {
+    console.log('Dataset was updated')
+    let oneDayBounds = getBounds(recentUpdated, 1)
+    let threeDaysBounds = getBounds(recentUpdated, 3)
+    let oneWeekBounds = getBounds(recentUpdated, 7)
+    let oneDayResults
+    let threeDaysResults
+    let oneWeekResults
+    try {
+      await Promise.all(
+        (oneDayResults = await getData(oneDayBounds)),
+        (threeDaysResults = await getData(threeDaysBounds)),
+        (oneWeekResults = await getData(oneWeekBounds))
+      )
+    } catch (e) {
+      e.code = 500
+      throw e
+    }
+    console.log('Adding new rows')
+    let id = uuid()
+    let newRows = {
+      id,
+      created_at: new Date(Date.now()).toISOString(),
+      last_updated: recentUpdated,
+      latest_water_level: recentWaterLevel,
+      one_day_average: oneDayResults[0].toFixed(3),
+      one_day_delta: oneDayResults[1].toFixed(3),
+      three_day_average: threeDaysResults[0].toFixed(3),
+      three_day_delta: threeDaysResults[1].toFixed(3),
+      one_week_average: oneWeekResults[0].toFixed(3),
+      one_week_delta: oneWeekResults[1].toFixed(3),
+      meta: {
+        id,
+        timestamp: Math.round(new Date() / 1000)
+      }
+    }
+    req.cache.add(KEY, newRows)
+    responseValues = await req.cache.getAll(KEY, req.body['limit'])
+  }
+
+  res.status(200).send({
+    data: responseValues
+  })
+}
+
+/**
+ * This function obtains the upper and lower bounds of a time period
+ * @param {String} date The date in ISO format
+ * @param {Number} days The number of days the period should be
+ * @returns {Array<String>} The upper and lower bounds of the time period
+ */
+function getBounds(date, days) {
+  const dateRegex = /[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}/
+  let currentDate = new Date(new Date(date) - TIMEZONE_OFFSET_MILLIS)
+  let threeDayPeriod = new Date(
```

Comment:
This variable name seems like its from a previous refactor.

Reply:
Please switch it to something like `oldDate` or something.

Reply:
Switched variable names to 'endOfPeriod' and 'startOfPeriod'